### PR TITLE
Migrate most testsets to test suites.

### PR DESCRIPTION
**Pull request by anuraaga**

### CODE REVIEW

File: context/build.gradle.kts [hunk 1]
```diff
@@ -70,7 +70,7 @@ testing {
 }
 
 tasks {
-  named("check") {
```
Comment (Oberon00) [inline context/build.gradle.kts:73]: I think this should be kept as-is to profit from conviguration-avoidance: https://docs.gradle.org/current/userguide/task_configuration_avoidance.html#sec:old_vs_new_configuration_api_overview

Comment (anuraaga) [reply context/build.gradle.kts:73]: It's not clear from the docs but it seems the kotlin dsl for these seems to use the lazy configuration APIs based on code navigating. Switched for the compile time safety

File: exporters/otlp/trace/build.gradle.kts [hunk 2]
```diff
@@ -53,8 +27,50 @@ dependencies {
...
+    }
+    val testGrpcOkhttp by registering(JvmTestSuite::class) {
+      dependencies {
+        implementation(project(":exporters:otlp:testing-internal"))
+
+        implementation("io.grpc:grpc-okhttp")
+        implementation("io.grpc:grpc-stub")
+      }
+    }
+    val testSpanPipeline by registering(JvmTestSuite::class) {
```
Comment (Oberon00) [inline exporters/otlp/trace/build.gradle.kts:56]: I think if we do not use the result of registering, then the `by registering` syntax looks odd. Isn't there some `register(JvmTestSuide::class)` API that could be used instead?

Comment (anuraaga) [reply exporters/otlp/trace/build.gradle.kts:56]: `register` with a string is the alternative. I find it more consistent to just always use this form even when not using the variable, here we happen to not have any need for variable form but if we happened to mix the two patterns it is confusing I think.

**Review by jack-berg — APPROVED** [review]

**Pull request merged**